### PR TITLE
DOCS-#3201: Add zsh install hint

### LINKED ISSUE #3201: Pip: no matches found: modin[ray]
### System information
- **OSX 11.3 (Big Sur)**:
- **modin-0.10.0**
- **Python 3.7.10**:

### Describe the problem
Can install modin, can't install dependencies using commands from docs. 

### Source code / logs
```
~ » pip install modin[ray] 
zsh: no matches found: modin[ray]
~ » pip install modin[dask]
zsh: no matches found: modin[dask]
~ » pip install modin[all] 
zsh: no matches found: modin[all]
```

### PR DESCRIPTION
**Pull request by alexlenail**
## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3201
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -52,7 +52,8 @@ of the targets:
    pip install modin[ray] # Install Modin dependencies and Ray to run on Ray
    pip install modin[dask] # Install Modin dependencies and Dask to run on Dask
    pip install modin[all] # Install all of the above
```
Comment (vnlitvinov) [inline docs/index.rst:54]:
```suggestion
   pip install "modin[ray]" # Install Modin dependencies and Ray to run on Ray
   pip install "modin[dask]" # Install Modin dependencies and Dask to run on Dask
   pip install "modin[all]" # Install all of the above
```

and the change below would not be needed

**Review by vnlitvinov — COMMENTED** [review]
Thanks @alexlenail for the contribution!

Please follow the templates for PR description and commit message.

**Comment by alexlenail** [comment]
@vnlitvinov do you need me to run `flake8` and `black`?

**Comment by vnlitvinov** [comment]
> @vnlitvinov do you need me to run `flake8` and `black`?

@alexlenail those would be run by our CI automatically - they're doing so now. We strive to merge at all passing (but coverage now could be flaky), so please pay attention to the CI.

Please also address my other feedback :)

**Comment by alexlenail** [comment]
Is it possible to squash and merge these?

**Comment by vnlitvinov** [comment]
Squashing is no problem (we do merge and squash anyway), but [we require](https://modin.readthedocs.io/en/latest/contributing.html#certificate-of-origin) you do a format "sign off" on your commits. Unless you do that the CI is going to be red 😢 

Plus I made [a suggestion](https://github.com/modin-project/modin/pull/3202#discussion_r662910436) of modifying the documentation in a way that would work in any shell (`zsh` included) instead of _adding_ a separate command line for `zsh`.

**Comment by alexlenail** [comment]
I have included your suggestion (sorry I didn't see it before!) and done a `git commit --amend --signoff`

**Comment by vnlitvinov** [comment]
Cool! I've squashed your commits now (because we require _all_ commits to be signed off, so I just squashed them to a single one), hope you don't mind that.

**Review by vnlitvinov — DISMISSED** [review]
LGTM

**Review by vnlitvinov — APPROVED** [review]

**Comment by vnlitvinov** [comment]
Thanks @alexlenail and sorry for that taking so long, I hope we've fixed the flakiness in our CI for the good!

**Pull request merged**